### ISSUE TITLE
Build tests on AppVeyor

### ISSUE THREAD

**Issue by hmenke** [hmenke/boost_matheval#1]
To be really useful we have to ensure that Boost Matheval builds on all platforms.

**Comment by hmenke** [comment]
Partially fixed in 02de3a27933ffc2f4ec91ef804c8222e4b45c38a.

Building tests still fails including but not limited to these errors
```
C:\Libraries\boost_1_59_0\boost/test/impl/test_tools.ipp(397): error C2491: 'boost::test_tools::tt_detail::prod_report_format': definition of dllimport function not allowed
C:\Libraries\boost_1_59_0\boost/test/tools/old/impl.hpp(117): error C2375: 'boost::test_tools::tt_detail::equal_impl': redefinition; different linkage
C:\Libraries\boost_1_59_0\boost/test/tools/old/impl.hpp(118): error C2264: 'boost::test_tools::tt_detail::equal_impl': error in function definition or declaration; function not called
```

**Comment by hmenke** [comment]
Now it somehow builds on the [appveyor branch](https://github.com/hmenke/boost_matheval/tree/appveyor) (I don't know why) but the tests don't run.  They fail with `***Exception: Other`
https://ci.appveyor.com/project/hmenke/boost-matheval/branch/appveyor

**Comment by hmenke** [comment]
A solution was found with the help of the Stack Overflow user sehe: https://stackoverflow.com/questions/47168753

Entirely fixed in 5e780397bc4cde6941295474d788c8cd70e1af77
